lib/wallet/src/Cardano/Wallet/Transaction/Delegation.hs

```diff
@@ -56,11 +63,13 @@ certificateFromDelegationAction
         -- ^ Era in which we create the certificate
     -> Either XPub (Script KeyHash)
         -- ^ Our staking credential
+    -> Maybe Coin
+        -- ^ Optional deposit value
```

Comment:
```
    -> Coin
        -- ^ Deposit value
```

Ideally, we want this to be `Coin` rather than `Maybe Coin` — the deposit is always known from the protocol parameters. However, fixing this in the current codebase is hard — the best approach would be to carry around the protocol parameters around all the way down `mkUnsignedTransaction` or even further.

Reply:
I will leave it for separate PR